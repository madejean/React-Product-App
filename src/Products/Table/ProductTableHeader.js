import React from 'react';

class ProductTableHeader extends React.Component {
  constructor(props){
    super(props);
    this.handleSort = this.handleSort.bind(this);
  }
  handleSort(e){
    this.props.onSort(this.props.columnHeader, e.target.name)
  }
  render() {
    var currentSortOption = this.props.currentSortOption.columnHeader === this.props.columnHeader ? this.props.currentSortOption.direction : false;
    return(
      <th>
        {this.props.columnHeader}
        <button
          className={currentSortOption === 'asc' ? 'Selected-sort-option' : ''}
          name='asc'
          onClick={this.handleSort}
        >&#x25B2;</button>
        <button
          className={currentSortOption === 'desc' ? 'Selected-sort-option' : ''}
          name='desc'
          onClick={this.handleSort}
        >&#x25BC;</button>
      </th>
    );
  }
}

export default ProductTableHeader;
