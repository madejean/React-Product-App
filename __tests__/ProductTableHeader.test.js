import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ProductTableHeader from '../src/Products/Table/ProductTableHeader';

test('ProductTableHeader component should render as expected', () => {
  var currentSortOption = {
    sort: {
    columnHeader: 'name',
    direction: 'desc'
    }
  }
  const component = shallow(<ProductTableHeader currentSortOption={currentSortOption} columnHeader="price" />)
  const tree = toJson(component)
  expect(tree).toMatchSnapshot()
})
