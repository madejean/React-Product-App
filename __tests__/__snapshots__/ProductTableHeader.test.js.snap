exports[`test ProductTableHeader component should render as expected 1`] = `
<th>
  price
  <button
    className=""
    name="asc"
    onClick={[Function]}>
    ▲
  </button>
  <button
    className=""
    name="desc"
    onClick={[Function]}>
    ▼
  </button>
</th>
`;
